.github/workflows/rust.yml

```diff
@@ -45,11 +45,69 @@ jobs:
         with:
           command: fmt
           args: --all -- --check
+      - if: failure()
```

Comment:
can you explain what this does?

Reply:
The intent behind this is to terminate any jobs that are running in parallel if one of them fails.

For example, if clippy fails first then we should terminate the check, wasm_check and docs steps. Unfortunately, it seems that this can only be achieved by making an API call. Ive replaced this code with [cancel-action](https://github.com/andymckay/cancel-action) script.

Reply:
Looks much better with the small 2 liner step (and the linked script looks ok)! We may want to use a hash version for the script to lock us to the specific code (as has been done in some other places) just in case!